doc/developers/develop.rst

```diff
@@ -496,6 +501,16 @@ no_validation (default=False)
 non_deterministic (default=False)
     whether the estimator is not deterministic given a fixed ``random_state``
 
+pairwise (default=False)
+    This boolean attribute indicates whether the data (`X`) :term:`fit` and
+    similar methods consists of pairwise measures over samples rather than a
+    feature representation for each sample.  It is usually `True` where an
+    estimator has a `metric` or `affinity` or `kernel` parameter with value
+    'precomputed'. Its primary purpose is that when a :term:`meta-estimator`
```

Comment:
when using cross-validation or when a meta-estimator?

I don't think GridSearchCV is the first meta-estimator people think of and I think cross_val_score etc are an important use-case.